In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [2]:
# importing data
player_data_url <- "https://www.dropbox.com/scl/fi/q73bfuafs2iasichsn736/players.csv?rlkey=xjsen4ho5dcb37m197u57neal&st=fvpz66cq&dl=1"
session_data_url = "https://www.dropbox.com/scl/fi/t6950eniqg71pxxwelyju/sessions.csv?rlkey=uj9mpdttf453ncujhj3yauymh&st=n53xekat&dl=1"

player_data <- read_csv(player_data_url)
session_data <- read_csv(session_data_url)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [3]:
# Data Description For player.csv (player_data)

# number of observations
num_obs_player <- nrow(player_data) #196 obs

# number of variables, name and type of variables, what the variables mean,
head(player_data)

#summary statistics
summary(player_data)

sd(player_data$Age, na.rm = TRUE)
sd(player_data$played_hours)

table(player_data$subscribe)
table(player_data$gender)
table(player_data$experience)

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

[1] 7.389687

[1] 28.35734


FALSE  TRUE 
   52   144 


          Agender            Female              Male        Non-binary 
                2                37               124                15 
            Other Prefer not to say      Two-Spirited 
                1                11                 6 


 Amateur Beginner      Pro  Regular  Veteran 
      63       35       14       36       48 

**Data description for player.csv (player_data)**
- Includes 196 observations, which means 196 unique players played on this Minecraft Server (PLAICraft)
- Includes 7 variables:
    1. `experience` (type **character**, one of):
        - Information on that player's experience playing Minecraft
            - 'Beginner'
            - 'Amateur'
            - 'Regular'
            - 'Veteran'
            - 'Pro'
    2. `subscribe` (type **logical**, one of):
      - Whether or not that player is subscribed
          - TRUE
          - FALSE
    3. `hashedEmail` (type **character**):
        - Represents the player's hashed email
    4. `played_hours` (type **double**):
        - Represents the player's total played hours on the server
    5. `name` (type **character**):
        - Represents the player's name
    6. `gender` (type **character**, one of):
        - Represents the player's gender
            - 'Male'
            - 'Female'
            - 'Non-binary'
            - 'Prefer not to say'
            - 'Agender'
            - 'Two-Spirited'
            - 'Other'
    7. `Age` (type **double**):
        - Represents the player's age
- Summary Statistics:
    - `Age` ranges from 9.00 to 58.00 years (mean = 21.14, median = 19.00, standard deviation = 7.39)
    - `played_hours` ranges from 0.00 to 223.10 hours (mean = 5.85, median = 0.10, standard deviation = 28.36)
    - Majority of the player's `gender` are 'Male' (124 out of 196)
    - Majority of the players's `experience` are 'Amateur' (63 players)
    - Majority of the players are subscribed (144 out of 196)
- Issues identified about the dataset:
    - There are 2 missing values in `Age`
    - It is not known:
        - how the data were collected
        - if it includes every player in a population or just a sample
  

**Questions:**

Broad Question: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

My Question: Can the player's experience, gender, age, and played hours predict whether the player will `subscribe` in the player_data dataset?

This dataset (player_data) contains many variables (player's experience, gender, age, and played hours) that can be used to predict a player's subscription status. Additionally, these variables provide predictor features (`experience`, `gender`, `Age`, `played_hours`) and target variable (`subscribe`) which means it can be used to build a classification model.

Plan for wrangling data:
- Since I only need `experience`, `gender`, `Age`, `played_hours`, `subscribe` variables, I can remove irrelevant variables (`name`, `hashedEmail`)
- Handle the missing values in `Age`
- Standardize ...

After cleaning the data, I can apply a classification method to predict `subscribe` based off the player's demographic features.


In [4]:
# Data Description For player.csv

# number of observations
num_obs_player <- nrow(player_data) #196 obs
num_obs_session <- nrow(session_data) #1535 obs

# number of variables, name and type of variables, what the variables mean,
head(player_data)
table(player_data$gender)
unique(player_data$Age)
#summary statistics
player_data_summary <- summary(player_data, digits = 3)
session_data_summary <- summary(session_data, digits = 3)
player_data_summary
#session_data_summary

test <- session_data |>
    mutate(new = (original_end_time - original_start_time))
#test
player_data_select <- player_data |>
    select(experience, played_hours, gender, Age)
#player_data_select
session_data_mutated <- session_data |>
    mutate("time_played", as.integer(substring(end_time, 11, 16)) - as.integer(substring(start_time, 11, 16)))
#sessions_data_mutated


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17


          Agender            Female              Male        Non-binary 
                2                37               124                15 
            Other Prefer not to say      Two-Spirited 
                1                11                 6 

[1]  9 17 21 19 47 22 23 25 58 28 18 33 26 20 24 27 42 11 10 38 14 15 37 16 12
[26] 46 32 49 45 NA 50 44 57

  experience        subscribe       hashedEmail         played_hours   
 Length:196         Mode :logical   Length:196         Min.   :  0.00  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.00  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.10  
                                                       Mean   :  5.85  
                                                       3rd Qu.:  0.60  
                                                       Max.   :223.10  
                                                                       
     name              gender               Age      
 Length:196         Length:196         Min.   : 9.0  
 Class :character   Class :character   1st Qu.:17.0  
 Mode  :character   Mode  :character   Median :19.0  
                                       Mean   :21.1  
                                       3rd Qu.:22.8  
                                       Max.   :58.0  
                                       NA's   

Warning message:
“There were 2 warnings in `mutate()`.
The first warning was:
ℹ In argument: `-...`.
Caused by warning:
! NAs introduced by coercion
ℹ Run `dplyr::last_dplyr_warnings()` to see the 1 remaining warning.”
